test/e2e/pod-alert_test.go

```diff
@@ -0,0 +1,382 @@
+package e2e_test
+
+import (
+	"github.com/appscode/go/crypto/rand"
+	"github.com/appscode/go/types"
+	tapi "github.com/appscode/searchlight/api"
+	"github.com/appscode/searchlight/test/e2e/framework"
+	. "github.com/appscode/searchlight/test/e2e/matcher"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+	apps "k8s.io/client-go/pkg/apis/apps/v1beta1"
+	extensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"strings"
+)
+
+var _ = Describe("PodAlert", func() {
+	var (
+		err             error
+		f               *framework.Invocation
+		rs              *extensions.ReplicaSet
+		ss              *apps.StatefulSet
+		pod             *apiv1.Pod
+		alert           *tapi.PodAlert
+		skippingMessage string
+	)
+
+	BeforeEach(func() {
+		f = root.Invoke()
+		rs = f.ReplicaSet()
+		ss = f.StatefulSet()
+		pod = f.Pod()
+		alert = f.PodAlert()
+		skippingMessage = ""
+	})
+
+	var (
+		shouldManageIcingaServiceForLabelSelector = func() {
+			By("Create ReplicaSet :" + rs.Name)
+			rs, err = f.CreateReplicaSet(rs)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Wait for Running pods")
+			f.EventuallyReplicaSet(rs.ObjectMeta).Should(HaveRunningPods(*rs.Spec.Replicas))
+
+			alert.Spec.Selector = *(rs.Spec.Selector)
+
+			By("Create matching podalert :"+ alert.Name)
+			err = f.CreatePodAlert(alert)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check icinga services")
+			f.EventuallyPodAlertIcingaService(alert.ObjectMeta, alert.Spec).
+				Should(HaveIcingaObject(IcingaServiceState{Ok: *rs.Spec.Replicas}))
+
+			By("Delete podalert")
+			err = f.DeletePodAlert(alert.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Wait for icinga services to be deleted")
+			f.EventuallyPodAlertIcingaService(alert.ObjectMeta, alert.Spec).
+				Should(HaveIcingaObject(IcingaServiceState{}))
+		}
+
+		shouldManageIcingaServiceForNewPod = func() {
+			By("Create ReplicaSet :" + rs.Name)
+			rs, err = f.CreateReplicaSet(rs)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Wait for Running pods")
+			f.EventuallyReplicaSet(rs.ObjectMeta).Should(HaveRunningPods(*rs.Spec.Replicas))
+
+			alert.Spec.Selector = *(rs.Spec.Selector)
+
+			By("Create matching podalert :"+ alert.Name)
+			err = f.CreatePodAlert(alert)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check icinga services")
+			f.EventuallyPodAlertIcingaService(alert.ObjectMeta, alert.Spec).
+				Should(HaveIcingaObject(IcingaServiceState{Ok: *rs.Spec.Replicas}))
+
+			rs, err = f.GetReplicaSet(rs.ObjectMeta)
```

Comment:
Use this update method
https://github.com/appscode/stash/blob/master/test/e2e/framework/replicaset.go#L41